aiidalab_widgets_base/nodes.py

```diff
@@ -210,22 +215,29 @@ def _find_called(cls, root):
     @classmethod
     def _find_outputs(cls, root):
         process_node = load_node(root.parent_pk)
-        if root.namespace:
-            outputs = process_node.outputs[root.namespace]
-        else:
-            outputs = process_node.outputs
 
+        # Gather outputs from node and its namespaces (if present):
+        outputs = functools.reduce(
+            lambda d, namespace: d[namespace],
+            root.namespaces or [],
+            process_node.outputs,
+        )
+
+        # Convert aiida.orm.LinkManager to dict
         output_nodes = {key: outputs[key] for key in outputs}
```

Comment:
What would happen if namespaces have outputs with the same key?

Reply:
This should already be validated by aiida, it is not allowed to have duplicate keys under one namespace.

Reply:
>  there can be no situation where outputs from different namespaces could share the same key ?

For the different namespaces, the keys can be the same. But the 'functools.reduce' unfold the namespace to the innermost nested specific namespace. For one namespace, keys can't be the same.